src/main/java/twilightforest/ASMHooks.java

```diff
@@ -108,6 +107,16 @@ public static boolean shouldMapRender(boolean o, ItemStack stack) {
 		return o || isOurMap(stack);
 	}
 
+	/**
+	 * Injection Point:<br>
+	 * {@link net.minecraft.world.entity.decoration.LeashFenceKnotEntity#survives()}<br>
+	 * [BEFORE IRETURN]
+	 */
+	public static boolean lead(boolean o, LeashFenceKnotEntity entity) {
+		BlockState fenceState = entity.level().getBlockState(entity.getPos());
+		return o || (fenceState.is(TFBlocks.WROUGHT_IRON_FENCE.get()) && fenceState.getValue(WroughtIronFenceBlock.POST) != Boolean.FALSE);
```

Comment:
Logic doesn't match Forge, is there a reason?
https://github.com/TeamTwilight/twilightforest/blob/1.20.1/src/main/java/twilightforest/ASMHooks.java#L333